### PR TITLE
[WebInterface] Check tentacles config on save

### PR DESCRIPTION
**Pull request by valouvaliavlo**

### CODE REVIEW

File: Services/Interfaces/web_interface/static/js/components/config_tentacle.js [hunk 1]
```diff
@@ -61,12 +61,25 @@ function handle_tentacle_config_update_error_callback(updated_data, update_url,
...
 function handleConfigDisplay(){
 
     if(canEditConfig()){
         $("#saveConfigFooter").show();
         $("#saveConfig").click(function() {
-            updateTentacleConfig(configEditor.getValue());
+            if(check_config())
+                updateTentacleConfig(configEditor.getValue());
+            else
+                alert("Invalid data form.")
```
Comment (GuillaumeDSM) [inline Services/Interfaces/web_interface/static/js/components/config_tentacle.js:82]:
```suggestion
                create_alert("error", "Error when saving configuration", "Invalid configuration data.");
```

File: Services/Interfaces/web_interface/static/js/components/config_tentacle.js [hunk 2]
```diff
@@ -61,12 +61,25 @@ function handle_tentacle_config_update_error_callback(updated_data, update_url,
     create_alert("error", "Error when updating config", msg.responseText);
 }
 
+function check_config(){
+    var errors = configEditor.validate();
+
+    if(errors.length)
+        return false
+    else
+        return true
```
Comment (GuillaumeDSM) [inline Services/Interfaces/web_interface/static/js/components/config_tentacle.js:70]:
```suggestion
    const errors = configEditor.validate();
    return !errors.length;
```
We prefer using const/let declaration in those js file to take advantage of modern js syntax

**Review by GuillaumeDSM — COMMENTED** [review]
very good idea, thanks !

File: Services/Interfaces/web_interface/static/js/components/config_tentacle.js (hunk 1 above)
Comment (GuillaumeDSM) [reply Services/Interfaces/web_interface/static/js/components/config_tentacle.js:82]: image

File: Services/Interfaces/web_interface/static/js/components/config_tentacle.js (hunk 2 above)
Comment (valouvaliavlo) [reply Services/Interfaces/web_interface/static/js/components/config_tentacle.js:70]: 👍

File: Services/Interfaces/web_interface/static/js/components/config_tentacle.js [hunk 3]
```diff
@@ -61,12 +61,26 @@ function handle_tentacle_config_update_error_callback(updated_data, update_url,
...
 }
 
+function check_config(){
+    var errors = configEditor.validate();
+
+    if(errors.length)
+        return false
+    else
+    const errors = configEditor.validate();
+    return !errors.length;
```
Comment (GuillaumeDSM) [inline Services/Interfaces/web_interface/static/js/components/config_tentacle.js:71]: I'm not sure I understand why calling twice `configEditor.validate()` ?
To me this 2 lines should be enough to validate the form and return a boolean corresponding to whether there are errors or not (basically the whole `check_config` function):
```python
const errors = configEditor.validate();    
return !errors.length;
```

File: Services/Interfaces/web_interface/static/js/components/config_tentacle.js [hunk 4]
```diff
@@ -61,12 +61,26 @@ function handle_tentacle_config_update_error_callback(updated_data, update_url,
...
 function handleConfigDisplay(){
 
     if(canEditConfig()){
         $("#saveConfigFooter").show();
         $("#saveConfig").click(function() {
-            updateTentacleConfig(configEditor.getValue());
+            if(check_config())
+                updateTentacleConfig(configEditor.getValue());
+            else
+                create_alert("error", "Error when saving configuration", "Invalid configuration data.");
```
Comment (GuillaumeDSM) [inline Services/Interfaces/web_interface/static/js/components/config_tentacle.js:83]: 👍

File: Services/Interfaces/web_interface/static/js/components/config_tentacle.js (hunk 3 above)
Comment (valouvaliavlo) [reply Services/Interfaces/web_interface/static/js/components/config_tentacle.js:71]: Sorry, I just applied the code review without looking

**Review by GuillaumeDSM — APPROVED** [review]
👍

**Review by Herklos — APPROVED** [review]

**Pull request merged**